File: src/main/java/io/github/shirohoo/library/application/persistance/book/BookJpaRepository.java
package io.github.shirohoo.library.application.persistance.book;

import io.github.shirohoo.library.domain.book.Book;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface BookJpaRepository extends JpaRepository<Book, Long> {
    Optional<Book> findByTitle(String bookTitle);
}

File: src/main/kotlin/io/github/shirohoo/library/application/persistence/book/BookJpaRepository.kt
package io.github.shirohoo.library.application.persistence.book

import io.github.shirohoo.library.domain.book.Book
import org.springframework.data.jpa.repository.JpaRepository

interface BookJpaRepository : JpaRepository<Book, Long> {
    fun findByTitle(bookTitle: String): Book?
}